### ISSUE TITLE
shuffle instabilities on color image with size 244x224 

### ISSUE THREAD

**Issue by wketcha** [keras-team/keras#13382]
I use 
- fashion MNIST 28x28x1 i resize to 224x224x3
- small network : flatten + 1 dense layer (128 neurons) + softmax (10 neurons).
- numpy, random, and tensorflow seed to get a reproducible shuflle.  

When i ran twice my training, i get different training results (different weights). Whereas, when i use Fashion MNIST 28x28x3 or 224x224x1, i get exactly stable results.

This is a bug in keras framework ?

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
